src/libraries/System.Security.Principal.Windows/tests/WindowsIdentityImpersonatedTests.netcoreapp.cs

```diff
@@ -0,0 +1,177 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.ComponentModel;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Principal;
+using System.Threading.Tasks;
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+// On nano server netapi32.dll is not present
+// we'll skip all tests on that platform
+public class WindowsIdentityImpersonatedTests : IClassFixture<WindowsIdentityFixture>
+{
+    private readonly WindowsIdentityFixture _fixture;
+
+    public WindowsIdentityImpersonatedTests(WindowsIdentityFixture windowsIdentityFixture)
+    {
+        _fixture = windowsIdentityFixture;
+
+        Assert.False(_fixture.TestAccount.AccountTokenHandle.IsInvalid);
+        Assert.False(string.IsNullOrEmpty(_fixture.TestAccount.AccountName));
+    }
+
+    [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
+    public async Task RunImpersonatedAsync_TaskAndTaskOfT()
+    {
+        WindowsIdentity currentWindowsIdentity = WindowsIdentity.GetCurrent();
+
+        await WindowsIdentity.RunImpersonatedAsync(_fixture.TestAccount.AccountTokenHandle, async () =>
+        {
+            Asserts(currentWindowsIdentity);
+            await Task.Delay(100);
+            Asserts(currentWindowsIdentity);
+        });
+
+        Assert.Equal(WindowsIdentity.GetCurrent().Name, currentWindowsIdentity.Name);
+
+        int result = await WindowsIdentity.RunImpersonatedAsync(_fixture.TestAccount.AccountTokenHandle, async () =>
+        {
+            Asserts(currentWindowsIdentity);
+            await Task.Delay(100);
+            Asserts(currentWindowsIdentity);
+            return 42;
+        });
+
+        Assert.Equal(42, result);
+        Assert.Equal(WindowsIdentity.GetCurrent().Name, currentWindowsIdentity.Name);
+
+        // Assertions
+        void Asserts(WindowsIdentity currentWindowsIdentity)
+        {
+            Assert.Equal(_fixture.TestAccount.AccountName, WindowsIdentity.GetCurrent().Name);
+            Assert.NotEqual(currentWindowsIdentity.Name, WindowsIdentity.GetCurrent().Name);
+        }
+    }
+}
+
+public class WindowsIdentityFixture : IDisposable
+{
+    public WindowsTestAccount TestAccount { get; private set; }
+
+    public WindowsIdentityFixture()
+    {
+        TestAccount = new WindowsTestAccount("CorFxTstWiIde01kiu");
+        TestAccount.Create();
```

Comment:
Any tests that create user accounts should be outerloop.

Reply:
added outerloop